scripts/compareAPIs.js

```diff
@@ -2,360 +2,545 @@
+    return 'never';
+  }
+  if (value.type === 'this') {
+    return 'this';
+  }
+  if (value.type === 'symbol') {
+    return 'symbol';
+  }
+  if (value.type === 'identifier') {
+    return value.name;
+  }
+  if (value.type === 'string') {
+    if (value.value) {
+      return `'${value.value}'`;
+    }
+    return 'string';
+  }
+  if (value.type === 'number') {
+    return 'number';
+  }
+  if (value.type === 'boolean') {
+    return 'boolean';
+  }
+  if (value.type === 'union') {
+    return value.elements.map(processType).join(' | ');
+  }
+  if (value.type === 'intersection') {
+    return `(${value.types.map(processType).join(' & ')})`;
+  }
+  if (value.type === 'application') {
+    let name = value.base.name;
+    if (!name) {
+      name = processType(value.base);
+    }
+    return `${name}<${value.typeParameters.map(processType).join(', ')}>`;
+  }
+  if (value.type === 'typeOperator') {
+    return `${value.operator} ${processType(value.value)}`;
+  }
+  if (value.type === 'function') {
+    return `(${value.parameters.map(processType).join(', ')}) => ${value.return ? processType(value.return) : 'void'}`;
+  }
+  if (value.type === 'parameter') {
+    return processType(value.value);
+  }
+  if (value.type === 'link') {
+    let name = value.id.substr(value.id.lastIndexOf(':') + 1);
+    if (dependantOnLinks.has(currentlyProcessing)) {
+      let foo = dependantOnLinks.get(currentlyProcessing);
+      if (!foo.includes(name)) {
+        foo.push(name);
+      }
+    } else {
+      dependantOnLinks.set(currentlyProcessing, [name]);
+    }
+    return name;
+  }
+  // interface still needed if we have it at top level?
+  if (value.type === 'object') {
+    if (value.properties) {
+      return `${value.exact ? '{\\' : '{'}
+  ${Object.values(value.properties).map(property => {
+    depth += 2;
+    let result = ' '.repeat(depth);
+    result = `${result}${property.indexType ? '[' : ''}${property.name}${property.indexType ? `: ${processType(property.indexType)}]` : ''}${property.optional ? '?' : ''}: ${processType(property.value)}`;
+    depth -= 2;
+    return result;
+  }).join('\n')}
+${value.exact ? '\\}' : '}'}`;
+    }
+    return '{}';
+  }
+  if (value.type === 'alias') {
+    return processType(value.value);
+  }
+  if (value.type === 'array') {
+    return `Array<${processType(value.elementType)}>`;
+  }
+  if (value.type === 'tuple') {
+    return `[${value.elements.map(processType).join(', ')}]`;
+  }
+  if (value.type === 'typeParameter') {
+    let typeParam = value.name;
+    if (value.constraint) {
+      typeParam = typeParam + ` extends ${processType(value.constraint)}`;
+    }
+    if (value.default) {
+      typeParam = typeParam + ` = ${processType(value.default)}`;
+    }
+    return typeParam;
+  }
+  if (value.type === 'component') {
+    let props = value.props;
+    if (props.type === 'application') {
+      props = props.base;
+    }
+    if (props.type === 'link') {
+      // create links provider
```

Comment:
Why do we have this `if` when it does nothing?

Reply:
this is something i want to address in followup, I don't think it's currently being hit, but it matches with the other file that I mentioned in the comment